core/src/test/scala/org/apache/spark/scheduler/DAGSchedulerSuite.scala

```diff
@@ -3709,8 +3709,8 @@ class DAGSchedulerSuite extends SparkFunSuite with TempLocalSparkContext with Ti
     assertDataStructuresEmpty()
   }
 
-  test("SPARK-32920: Merge results should be unregistered if the running stage is cancelled" +
-    " before shuffle merge is finalized") {
+  test("SPARK-32920: Merge results should not be unregistered if a determinate running stage is " +
+    "cancelled before shuffle merge is finalized") {
```

Comment:
nit: we only want the merge results not to be unregistered in the case of stage retry but for cancelled stage it is fine to remove it right?. I don't think any other stages would be able to consume it even in the case of determinate stage.

But I think we don't distinguish between stage retry and cancellation here (or too tedious to do for nothing). I feel we can rephrase the test name to reflect accordingly.

Reply:
> we only want the merge results not to be unregistered in the case of stage retry but for cancelled stage it is fine to remove it right?
We are unable to differentiate between a cancelled stage or a stage that will be re-attempted in `handleShuffleMergeFinalized`. So, for a determinate stage we don't unregister the merge results.
I will rephrase the test name